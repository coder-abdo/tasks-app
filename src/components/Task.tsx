import React from "react";
import { FaTag, FaTimes } from "react-icons/fa";
import { AppTask } from "../styles/TaskStyle";
import { IProps } from "../interfaces/interfaces";

const Task = ({ title, cateogry, handleRemoveTask }: IProps) => {
  return (
    <AppTask data-testid="task">
      <h2 className="task-title">{title}</h2>
      <div className="cateogry">
        <FaTag />
        <span>category: {cateogry}</span>
      </div>
      <FaTimes onClick={handleRemoveTask} className="remove-btn" />
    </AppTask>
  );
};

export default Task;
