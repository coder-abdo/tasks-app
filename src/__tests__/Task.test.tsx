import React from "react";
import Task from "../components/Task";
import {
  cleanup,
  render,
  fireEvent,
  getByTestId
} from "@testing-library/react";
beforeEach(cleanup);

describe("<Task />", () => {
  it("render Task Component", () => {
    const mockData = {
      id: "1",
      title: "learn testing",
      cateogry: "studying",
      handleRemoveTask: jest.fn()
    };
    const { queryByTestId } = render(<Task {...mockData} />);
    expect(queryByTestId("task")).toBeTruthy();
    expect(queryByTestId("task")).toMatchSnapshot();
  });
});
