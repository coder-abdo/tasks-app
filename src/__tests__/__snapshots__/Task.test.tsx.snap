// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Task /> render Task Component 1`] = `
<div
  class="sc-bdVaJa ehhtcL"
  data-testid="task"
>
  <h2
    class="task-title"
  >
    learn testing
  </h2>
  <div
    class="cateogry"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 252.118V48C0 21.49 21.49 0 48 0h204.118a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882L293.823 497.941c-18.745 18.745-49.137 18.745-67.882 0L14.059 286.059A48 48 0 0 1 0 252.118zM112 64c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48z"
      />
    </svg>
    <span>
      category: 
      studying
    </span>
  </div>
  <svg
    class="remove-btn"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 352 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
    />
  </svg>
</div>
`;
